engine/verification/fetcher/engine.go

```diff
@@ -501,12 +511,12 @@ func (e *Engine) pushToVerifier(chunk *flow.Chunk,
 	chunkDataPack *flow.ChunkDataPack,
 	collection *flow.Collection) error {
 
-	header, err := e.headers.ByBlockID(chunk.BlockID)
+	block, err := e.blocks.ByID(chunk.BlockID)
```

Comment:
here it's better to query by header, so that we don't need to query the block payload. 

So we could import both headers and blocks. And use headers here and use blocks when looking for collection.

Reply:
Applied in https://github.com/onflow/flow-go/pull/737/commits/e33df4c18caab8e345cca6ba1b0be9298f36f2e3.